source/acceptance_test.go

```diff
@@ -0,0 +1,215 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package source_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+
+	marketo "github.com/rustiever/conduit-connector-marketo"
+	"github.com/rustiever/conduit-connector-marketo/source"
+	"github.com/rustiever/conduit-connector-marketo/source/position"
+	"go.uber.org/goleak"
+)
+
+func TestAcceptance(t *testing.T) {
+	src := &source.Source{}
+	client, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sdk.AcceptanceTest(t, AcceptanceTestDriver{
+		Config: AcceptanceSourceTestDriverConfig{
+			Connector: sdk.Connector{
+				NewSpecification: marketo.Specification,
+				NewSource:        func() sdk.Source { return src },
+			},
+			SourceConfig: getConfigs(),
+			GoleakOptions: []goleak.Option{
+				goleak.IgnoreCurrent(),
+				goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+				// external dependency - minimarketo
+				goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+				goleak.IgnoreTopFunction("net/http.(*persistConn).roundTrip"),
+				goleak.IgnoreTopFunction("net/http.setRequestCancel.func4"),
+			},
+			BeforeTest: func(t *testing.T) {
+				// iterator.InitialDate = time.Now().UTC()
+				src.InitialDate = time.Now().UTC()
+			},
+			AfterTest: func(t *testing.T) {
+				t.Cleanup(func() {
+					err := cleanUp(client)
+					if err != nil {
+						t.Error(err)
+					}
+				})
+			},
+		},
+	},
+	)
+}
+
+// AcceptanceTestDriver implements sdk.AcceptanceTestDriver
+type AcceptanceTestDriver struct {
+	Config AcceptanceSourceTestDriverConfig
+	sdk.ConfigurableAcceptanceTestDriver
+}
+
+// AcceptanceSourceTestDriverConfig contains the configuration for
+// AcceptanceTestDriver.
+type AcceptanceSourceTestDriverConfig struct {
+	// Connector is the connector to be tested.
+	Connector sdk.Connector
+
+	// SourceConfig should be a valid config for a source connector, reading
+	// from the same location as the destination will write to.
+	SourceConfig map[string]string
+	// DestinationConfig should be a valid config for a destination connector,
+	// writing to the same location as the source will read from.
+	DestinationConfig map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest func(t *testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest func(t *testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+}
```

Comment:
@rustiever Sorry for the late response! I'd be grateful if you could remind what was the error you had, which forced you to embed `sdk.ConfigurableAcceptanceTestDriver` but also copy paste the fields and methods. Other than this, I see no blockers to merging this PR.

Reply:
refactored